src/wp-admin/includes/class-wp-plugins-list-table.php

```diff
@@ -210,6 +210,21 @@ public function prepare_items() {
 				$plugin_data['auto-update-supported'] = false;
 			}
 
+			// TODO: The slug isn't set for non-update-enabled plugins, but must be set for the filter.
+			if ( ! isset( $plugin_data['slug'] ) ) {
+				if ( '.' === dirname( $plugin_file ) ) {
+					$plugin_data['slug'] = basename( $plugin_file );
+				} else {
+					$plugin_data['slug'] = dirname( $plugin_file );
+				}
+			}
+
+			// TODO: Document, make sure we're passing the right data as $plugin_data.
+			$auto_update_forced = apply_filters( 'auto_update_plugin', null, (object) $plugin_data );
```

Comment:
Something similar is done in `WP_Site_Health::detect_plugin_theme_auto_update_issues`

Reply:
Thanks! that formed the basis of https://github.com/WordPress/wordpress-develop/pull/431/commits/9363ecabc05d4d9c5a62fc2ad894d9e0bcaea7ab

Reply:
The messages in Site Health are of the form `Auto-updates for plugins appear to be disabled...` (instead of `...are disabled...`) specifically because passing a mock item to the filter is not guaranteed to produce the same result as when a real item/offer is passed when the automatic updater is run for real.  Site Health is intended to alert site admins of _possible_ problems, which is a very different scenario than limiting the UI for auto-updates.